### PR TITLE
18860 2 nodes same license

### LINKED ISSUE #18860: 2 Nodes can pick the same license
There are two cases we need to fix here:

1. **If you have license pack, it is possible for 2 nodes to select the same license.**  This means that one of these servers does not register itself into the sitelic table.  If a server is not in the sitelic table, it will not be registered in dotCMS as a valid part of the cluster.  We need to avoid servers being able to pick the same license.
To test, bring up a cluster without licenses and upload a license pack.

2. **You can have two servers that share a serverId.** ServerIds should be unique to each dotCMS instance. Sometimes, when setting up a cluster, customers clone a server (including the `/dotsecure` folder) and point it at the same db, then two servers end up sharing the same Server Id.  This causes issue with the cluster as each server tries to take the same license and cannot register themselves. 

I think we need to add a column to the sitelic table which is called `startup_time` that records the value of:
`long jvmStartTime = ManagementFactory.getRuntimeMXBean().getStartTime();`.  In the process of updating the sitelic record, we should make sure to only update the sitelic record where the `startup_time` matches the JVMs `getStartTime()`.  If the server_ids match but the `getStartTime()`, then we know two servers are sharing a server_id and we should throw a warning in the logs.

### PR DESCRIPTION
**Pull request by victoralfaro-dotcms**
EE PR: https://github.com/dotCMS/enterprise-2.x/pull/758

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/license/LicenseManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package com.dotcms.enterprise.license;
+
+public class LicenseManagerTest {
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/license/LicenseManagerTest.java:3]: What is this empty class for?

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task201102UpdateColumnSitelicTableTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import com.dotmarketing.exception.DotDataException;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.sql.SQLException;
+
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class Task201102UpdateColumnSitelicTableTest {
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task201102UpdateColumnSitelicTableTest.java:15]: missing javadoc

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/license/LicenseManagerTest.java (hunk 1 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/integration-test/java/com/dotcms/enterprise/license/LicenseManagerTest.java]: Removed.

File: dotCMS/dependencies.gradle [hunk 3]
```diff
@@ -3,7 +3,8 @@ dependencies {
 
     def eeType = dotcmsReleaseVersion.equals("master") ? '-SNAPSHOT' : ''
 
-    compile group: 'com.dotcms.enterprise', name: 'ee', version: dotcmsReleaseVersion + eeType, changing: true
+    //compile group: 'com.dotcms.enterprise', name: 'ee', version: dotcmsReleaseVersion + eeType, changing: true
+    compile fileTree(dir: 'src/main/enterprise/build/libs', include: ['ee_clean.jar'])
```
Comment (freddyucv) [inline dotCMS/dependencies.gradle:7]: you have to remove this change

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task201102UpdateColumnSitelicTableTest.java (hunk 2 above)
Comment (freddyucv) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task201102UpdateColumnSitelicTableTest.java:15]: doc

File: .gitmodules [hunk 4]
```diff
@@ -1,5 +1,5 @@
 [submodule "dotCMS/src/main/enterprise"]
 	path = dotCMS/src/main/enterprise
 	url = git@github.com:dotCMS/enterprise-2.x.git
-        branch = master
+	branch = 18860-2-nodes-same-license
```
Comment (freddyucv) [inline .gitmodules:4]: is this change necessary?

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: .gitmodules (hunk 4 above)
Comment (victoralfaro-dotcms) [reply .gitmodules:4]: No it's just to make it work locally. Going to remove this change from PR.

File: dotCMS/dependencies.gradle (hunk 3 above)
Comment (victoralfaro-dotcms) [reply dotCMS/dependencies.gradle:7]: Yes, removed

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task201102UpdateColumnSitelicTableTest.java (hunk 2 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task201102UpdateColumnSitelicTableTest.java:15]: Added.

**Review by freddyucv — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]
